_blog/2018/soft-multitenancy.md

```diff
@@ -0,0 +1,298 @@
+subtitle: Using multiple Istio control planes and RBAC to create multitenancy
+attribution: John Joyce and Rich Curran
+
+order: 92
+
+layout: blog
+type: markdown
+redirect_from: "/blog/soft-multitenancy.html"
+---
+{% include home.html %}
+Multitenancy is commonly used in many environments across many different applications,
+but the implementation details and functionality provided on a per tenant basis does not
+follow one model in all environments.  The Kubernetes multitenancy [working group](
+https://github.com/kubernetes/community/blob/master/wg-multitenancy/README.md)
+is working to define the multitenant use cases and functionality that should be available
+within Kubernetes. From their work so far it is clear that only a "soft multitenancy" is
+possible due to the inability to fully protect against malicious containers or workloads
+gaining access to the other pods or kernel resources.
+
+## Defining Soft Multitenancy for Istio
+While discussing the strengths and benefits of deploying applications on top of Istio it
+became apparent that there are multitenant use cases for Istio even within this "soft"
+environmental that does not provide absolute protection amongst tenants.  A viable use case
+for this scenario is shared corporate infrastructure where malicious actions are not
+expected but a clean separation of the tenants is still required. A couple different
+multitenant models could be considered.
+1.	A single mesh with multiple applications, one for each tenant on the mesh. The cluster
+administrator gets control and visibility mesh wide and across all applications, while the
+tenant administrator only gets control of a specific application.
+1.	A single Istio control plane with multiple meshes, one mesh per tenant. The cluster
+administrator gets control and visibility across the entire Istio control plane and all
+meshes, while the tenant administrator only gets control of a specific mesh.
+1.	A single Kubernetes control plane with multiple Istio control planes, one per tenant.
+The cluster administrator gets control and visibility across all the Istio control planes,
+while the tenant administrator only gets control of a specific Istio instance.
+1.	A single cloud environment (cluster controlled), but multiple kubernetes control planes
+(tenant controlled).
+
+The fourth model doesn’t satisfy most use cases, as most cluster administrators prefer
+a common kubernetes control plane with which they provide as a PaaS to their tenants.
+Additionally, case 4 is easily provided in many environments already.
+
+Current Istio capabilities are poorly suited to support the first model as it lacks
+sufficient RBAC capabilities to support cluster versus tenant operations. Additionally,
+having multiple tenants under one mesh is too insecure with the current mesh model.
+So option 1 won’t be further evaluated in this blog.
+
+Regarding the 2nd option the current Istio paradigm assumes a single mesh per Istio control
```

Comment:
This is definitely going to happen as a cost saving approach to scaling an Istio-based PasS business. We're trying to be account for this model when designing Istio technologies, such that we'll be able to effectively take all of the Istio goo out of the individual customer clusters and house that stuff in a centralized location. Thus, Istio components need to be designed for true multitenancy.